src/main/java/mate/academy/dao/OrderDao.java

```diff
@@ -0,0 +1,14 @@
+package mate.academy.dao;
+
+import java.util.List;
+import java.util.Optional;
+import mate.academy.model.Order;
+import mate.academy.model.User;
+
+public interface OrderDao {
+    Order add(Order order);
+
+    Optional<Order> getByUser(User user);
+
+    List<Order> getOrdersByUser(User user);

```

Comment:
The methods `getByUser` and `getOrdersByUser` seem to overlap in functionality. `getByUser` returns an `Optional<Order>`, while `getOrdersByUser` returns a `List<Order>`. If the intention is to retrieve multiple orders for a user, `getOrdersByUser` should suffice. Consider removing `getByUser` if it is not necessary, or clarify the distinction between these two methods.